xds/internal/client/client_xds.go

```diff
@@ -405,10 +407,83 @@ func validateCluster(cluster *v3clusterpb.Cluster) (ClusterUpdate, error) {
 		return emptyUpdate, fmt.Errorf("xds: unexpected lbPolicy %v in response: %+v", cluster.GetLbPolicy(), cluster)
 	}
 
-	return ClusterUpdate{
+	cu := ClusterUpdate{
 		ServiceName: cluster.GetEdsClusterConfig().GetServiceName(),
 		EnableLRS:   cluster.GetLrsServer().GetSelf() != nil,
-	}, nil
+	}
+	sc, err := securityConfigFromCluster(cluster)
+	if err != nil {
+		return emptyUpdate, err
+	}
+	if sc != nil {
+		// A valid Cluster resource could contain no security configuration
+		// (transport_socket field is nil). Hence we need a nil check here.
+		cu.SecurityCfg = *sc
```

Comment:
How would the CDS balancer know if this field exists in the CDS response? Does it check for empty strings? Or make this field a pointer?

Reply:
Done. Changed it to a pointer. I initially had it as a pointer, then saw that none of the `Update` types had a pointer, so switched it to value. But pointer definitely is better here. Thanks.